impl/src/main/java/org/pentaho/platform/genericfile/providers/repository/RepositoryFileProvider.java

```diff
@@ -360,6 +360,22 @@ private Date getModifiedDateFromNativeFileDto( @NonNull RepositoryFileDto native
     return null;
   }
 
+  @Nullable
+  private Date getModifiedDateFromNativeFileDto( @NonNull RepositoryFileDto nativeFile ) {
+    Date lastModified = getDate( nativeFile.getLastModifiedDate() );
+    if ( lastModified != null ) {
+      return lastModified;
+    }
+
+    return getDate( nativeFile.getCreatedDate() );
+  }
+
+  private String getOwnerFromNativeFileDto( @NonNull RepositoryFileDto nativeFile ) {
+    RepositoryFileAcl acl = unifiedRepository.getAcl( nativeFile.getId() );
+
+    return acl.getOwner().getName();
```

Comment:
Is there any issue with the data in `nativeFile.getOwner()`?

Reply:
The only issue was me totally not noticing that we could get the owner directly from `RepositoryFileDto` 😅

Reply:
It seems there are some issues:
1. for general use, we need to set `includeAcls` to `true`so that the owner is defined in `RepositoryFileDto`
1. for deleted files, we in fact need to access the owner this way, as the current platform service doesn't expose the original owner (and no flag to include it on demand)

Reply:
As discussed offline, that seems to be the correct approach. Only additional thing would be to add a comment justifying the approach.